presto-main/src/main/java/com/facebook/presto/cost/CachingStatsProvider.java

```diff
@@ -59,7 +58,7 @@ public CachingStatsProvider(StatsCalculator statsCalculator, Optional<Memo> memo
     public PlanNodeStatsEstimate getStats(PlanNode node)
     {
         if (!isEnableStatsCalculator(session)) {
-            return UNKNOWN_STATS;
+            return PlanNodeStatsEstimate.unknown();
```

Comment:
I would prefer `unknownNodeStats()` so you could use `import static`

Reply:
@arhimondr i admit i missed https://github.com/prestodb/presto/pull/11463#discussion_r217391682
if this is the direction we all / you all agree to go, then i am OK.
Today, I still have slight impression that we're changing from "someone else's preferred way" to "Andrii's preferred way" rather than to "the Presto way". This is not a huge deal to me anyway, but please make really really sure that no refactoring commit like this introduces any semantics changes -- I don't like to review refactoring commits 😄

Reply:
Let me explain a bit why did i do this refactoring.

Originally i wanted to make `Optional<PlanNodeStatEstimate>` to be just `PlanNodeStatEstimate` in the `FilterStatsCalculator`. It was confusing how is `Optional.empty()` is different from uknown estimate. 

But once i tried to remove `Optional` from the methods signatures, i realize that instead of empty i need to return `PlanNodeStatEstimate.UNKNOWN_STATS`. I was introducing a name clash in static imports. So, static importing `UNKNOWN_STATS` was no longer an option. 

Using `PlanNodeStatEstimate.UNKNOWN_STATS` without static import is tautological, as it duplicates word `STATS`.

I could shortened `UNKNOWN_STATS` to `UNKNOWN`, but that would've introduced even higher change of a name clash in the static imports elsewhere.

Also even before the refactoring it wan't always obvious what exactly `UNKNOWN_STATS` is returned? Is it `PlanNodeStatEstimate` or `SymbolNodeStatEstimate`?

The other question is "Why a static constructor method is preferred over a constant?". In most of the cases this is just a matter of taste. But in theory having a method gives you a better flexibility, like being able to lazy initialize the constant in case it is ever needed.

Reply:
I agree with the linked comment.

In general, a static factory method seems preferable to a public constant, since it provides encapsulation (e.g., the implementation could change to be a computed value rather than a constant). Modern JDK APIs like `Optional` follow this pattern. However, this is orthogonal to the question of qualified name vs static import.

(For primitives and strings, there is a huge difference between a constant and a method: constants are inlined at compile time, so changing the constant will change clients if they are compiled separately. This is why you can reference constants in annotations but not static methods that return constants.)

A static import `emptyOptional()` feels a bit nicer to read than `Optional.empty()`, but can have the problem of ambiguity. Sometimes this is deliberate: you can shorten `PlanNodeStatEstimate.unknown()` to a nicer looking `unknownStats()`.

In the end, it has to be a judgement call on a case by case basis. For this case, given the preference for factory methods and the ambiguity of *unknown stats*, using a `unknown()` method seems like the best choice.

Reply:
Another argument for qualified names is that there is no question over whether or not to static import them. Hopefully, it's obvious to never import `Optional.empty()`. But based on experience, we would likely see many non-imported usages `Optional.emptyOptional()`.

As an example, consider how many usages we have of `UUID.randomUUID()` in our code base.

Reply:
> Originally i wanted to make Optional<PlanNodeStatEstimate> to be just PlanNodeStatEstimate in the FilterStatsCalculator. It was confusing how is Optional.empty() is different from uknown estimate.

@arhimondr the choice of `Optional.empty()` was deliberate, because otherwise it's very hard to say if stats are "unknown enough".
Equality check with `PlanNodeStatEstimate.UNKNOWN_STATS` is not a good idea, since there is no practical diffence between `PlanNodeStatEstimate.UNKNOWN_STATS` and stats where we know that some symbol has null fraction 0.5 (and nothing more). We used to have code like that, but it looked creepy and then we changed to signal "i don't know" with `Optional.empty()`

(@electrum  -- regarding `UUID.randomUUID()` vs `UUID.random()`: latter removes the inconsistency as it should not be static-imported; IMO, it removes also the convenience when writing, but maybe that's not a big deal)

Reply:
>  was deliberate, because otherwise it's very hard to say if stats are "unknown enough".

If the estimated row count is unknown, aren't the stats "uknown enough"?

Reply:
Depends on the context -- here, in FilterStatsCalculator I think yes, "unknown enough". In explain, this might not be "unknown enough" (although currently is)
However, if you want to make such a change (considering unkown row count as unknown enough), please make it a separate change from other changes.

Reply:
> In explain, this might not be "unknown enough" (although currently is)

I renamed the method to `isOutputRowCountUnknown` to avoid this confusion

> However, if you want to make such a change (considering unkown row count as unknown enough), please make it a separate change from other changes.

Just did. Sorry for making this mess. I should've extracted all the refactoring from the beginning.